File: LeetCode/p2730.py
class Solution(object):
    def longestSemiRepetitiveSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """

        l, max_size, same = 0, 1, 0

        for r in range(1, len(s)):
            if r > 0 and s[r] == s[r - 1]:
                same += 1

            while same > 1:
                if s[l + 1] == s[l]:
                    same -= 1
                l += 1

            max_size = max(max_size, r - l + 1)

        return max_size

File: LeetCode/p2730.cpp
#include <bits/stdc++.h>
using namespace std;
#define lint long long;
#define endl "\n"

class Solution {
public:
    int longestSemiRepetitiveSubstring(string s) {
        int maxSize = 1;
        const int size = s.size();
        int same = 0;
        int L = 0;
        for(int R = 0 ; R < size ; ++R)
        {
            if(R > 0 && s[R] == s[R-1])
                same++;

            while(same > 1)
            {
                if(s[L+1] == s[L] && L + 1 <= R)
                    same--;
                L++;
            }
            maxSize = max(maxSize, R-L+1);
        }

        return maxSize;
    }
};